### PR TITLE
Add data residency validation to webhook

### LINKED ISSUE #1741: Data residency configmap is not validated by webhook
**Describe the bug**
Followup from https://github.com/google/knative-gcp/pull/1681#issuecomment-695042860.

 **Expected behavior**
Invalid keys in the config-data-residency configmap cause a webhook error.

/assign @zhongduo

### PR DESCRIPTION
**Pull request by zhongduo**
Fixes #1741 

## Proposed Changes

- Add data residency configuration map to webhook validation

### CODE REVIEW

**Comment by zhongduo** [comment]
The fix to the unit test error will conflict with #1733, will fix it after rebase.

File: pkg/apis/configs/dataresidency/dataresidency_defaults.go [hunk 1]
```diff
@@ -50,10 +51,8 @@ func NewDefaultsConfigFromMap(data map[string]string) (*Defaults, error) {
 
 	// Parse out the data residency configuration.
 	value, present := data[defaulterKey]
-	// Data residency configuration is not required, in which case it will mean
-	// allow all regions, so we simply use an empty one
 	if !present || value == "" {
-		return nc, nil
+		return nil, fmt.Errorf("ConfigMap is missing (or empty) key: %q : %v", defaulterKey, data)
```
Comment (Harwayne) [inline pkg/apis/configs/dataresidency/dataresidency_defaults.go:55]: Are we sure we want this to be an error?

My main concern was typos and the `DisallowUnknownFields()` below should remove that possibility, right?

Comment (zhongduo) [reply pkg/apis/configs/dataresidency/dataresidency_defaults.go:55]: Not really. The configmap has a weird way of parsing the file, so this is literally only checking if the key `default-dataresidency-config` exists. The function `parseEntry` actually only parse the `value` part only, which starts from `clusterDefaults`. So this conditions is triggered only when the `default-dataresidency-config` key is not present, same as the `gcp-auth` behavior.

**Review by Harwayne — COMMENTED** [review]
/lgtm

**Pull request merged**